molgenis-data-postgresql/src/test/java/org/molgenis/data/postgresql/PostgreSqlExceptionTranslatorTest.java

```diff
@@ -55,10 +57,9 @@ public void translateReadonlyViolation()
 		when(serverErrorMessage.getMessage()).thenReturn(
 				"Updating read-only column \"myColumn\" of table \"myTable\" with id [abc] is not allowed");
 		//noinspection ThrowableResultOfMethodCallIgnored
-		MolgenisValidationException e = postgreSqlExceptionTranslator.translateReadonlyViolation(
+		DataIntegrityViolationException e = postgreSqlExceptionTranslator.translateReadonlyViolation(
 				new PSQLException(serverErrorMessage));
-		assertEquals(e.getMessage(),
-				"Updating read-only attribute 'myAttr' of entity 'myEntity' with id 'abc' is not allowed.");
+		assertEquals(e.getMessage(), "type:myEntity attribute:myAttr value:abc");
```

Comment:
The exact type of the exception is not being checked. Either do this with an equals() method or with standard testng expected exception + message annotation.

(this comment times n)

Reply:
The exception here is returned, not thrown, so the standard testNg stuff wont work. (unless we throw the returned exception, which would be a bit odd in this case).
It is certain that this is a DataIntegrityViolationException, since that is the return type of the method, this is an abstract class so we might want to check the exact class anyway